### PR TITLE
FM-54_change to M2M tag-deck relation with updated tests and example data f…

### PR DESCRIPTION
**Pull request by bartoszbaran2**
Changing the relationship to many-to-many between the deck model the tag model with updated tests for the models and a sample data generator

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

tested with unit tests and manually

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added

alt memme

### CODE REVIEW

File: flashcards/backend/decks/factories.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from factory import Faker, Sequence, SubFactory
 from factory.django import DjangoModelFactory, ImageField
+import factory
```
Comment (gonzur75) [inline flashcards/backend/decks/factories.py:3]: Why import whole factory?

Comment (bartoszbaran2) [reply flashcards/backend/decks/factories.py]: just automatic import
i've fixed. thanks for checking

**Review by gonzur75 — APPROVED** [review]
Good to go.

**Review by habibuga — APPROVED** [review]
Good job! :+1:

File: flashcards/backend/decks/models.py [hunk 2]
```diff
@@ -100,15 +100,15 @@ class Tag(models.Model):
...
+    :type decks: int
     :param flashcard: Foreign Key for Flashcard model
     :type flashcard: int
     """
 
     name = models.CharField(max_length=24)
-    deck = models.ForeignKey("Deck", on_delete=models.DO_NOTHING)
-    flashcard = models.ForeignKey("Flashcard", on_delete=models.DO_NOTHING)
+    decks = models.ManyToManyField("Deck", related_name="tags")
+    flashcard = models.ForeignKey("Flashcard", on_delete=models.DO_NOTHING, related_name="tags")
```
Comment (martynawitkowska) [inline flashcards/backend/decks/models.py:111]: The flashcard should also be a MM relation. while we are here I'm wondering, if adding `blank=True` to those fields to make them not required while creating a tag.

**Review by martynawitkowska — CHANGES_REQUESTED** [review]
I think it would be great to add an inline to deck model in admin site. This would maake all tags assigned to specific deck visible on deck details site in admin.

https://docs.djangoproject.com/en/4.2/ref/contrib/admin/#inlinemodeladmin-objects

**Pull request merged**